internal/agent/validator.go

```diff
@@ -0,0 +1,266 @@
+	}
+
+	for _, pattern := range suspiciousPatterns {
+		matched, _ := regexp.MatchString(pattern.pattern, contentStr)
+		if matched {
+			result.Errors = append(result.Errors, fmt.Sprintf("files: suspicious content in %s: %s", filename, pattern.message))
+		}
+	}
+
+	// Check for hardcoded secrets patterns
+	secretPatterns := []struct {
+		pattern string
+		message string
+	}{
+		{`(?i)(api[_-]?key|apikey)\s*[:=]\s*["\']?[a-zA-Z0-9]{20,}`, "potential API key"},
+		{`(?i)(secret|password|passwd|pwd)\s*[:=]\s*["\']?[a-zA-Z0-9]{8,}`, "potential hardcoded secret"},
+		{`(?i)token\s*[:=]\s*["\']?[a-zA-Z0-9]{20,}`, "potential access token"},
+		{`sk-[a-zA-Z0-9]{20,}`, "potential OpenAI API key"},
+		{`ghp_[a-zA-Z0-9]{36}`, "potential GitHub personal access token"},
+	}
+
+	for _, pattern := range secretPatterns {
+		matched, _ := regexp.MatchString(pattern.pattern, contentStr)
+		if matched {
+			result.Errors = append(result.Errors, fmt.Sprintf("files: potential hardcoded secret in %s: %s", filename, pattern.message))
+		}
+	}
+}
+
+func (v *AgentValidator) validateSecurity(pkg *AgentPackage, result *ValidationResult) {
+	// Check source security
+	if pkg.Source.Type == SourceTypeURL {
+		if !strings.HasPrefix(pkg.Source.Location, "https://") {
+			result.Errors = append(result.Errors, "source: only HTTPS URLs are allowed for security")
+		}
+	}
+
+	// Validate dependencies if present
+	if pkg.Metadata.Dependencies != nil {
+		v.validateDependencies(pkg.Metadata.Dependencies, result)
+	}
+}
+
+func (v *AgentValidator) validateDependencies(deps *AgentDependencies, result *ValidationResult) {
+	// Check for known malicious packages
+	maliciousPackages := map[string][]string{
+		"npm": {"event-stream", "eslint-scope", "getcookies"},
+		"pip": {"python3-dateutil", "python3-urllib3", "jeIlyfish"},
+	}
+
+	if deps.NPM != nil {
+		for _, pkg := range deps.NPM {
+			if v.isMaliciousPackage("npm", pkg, maliciousPackages["npm"]) {
+				result.Errors = append(result.Errors, fmt.Sprintf("dependencies: potentially malicious NPM package: %s", pkg))
+			}
+		}
+	}
+
+	if deps.Pip != nil {
+		for _, pkg := range deps.Pip {
+			if v.isMaliciousPackage("pip", pkg, maliciousPackages["pip"]) {
+				result.Errors = append(result.Errors, fmt.Sprintf("dependencies: potentially malicious Python package: %s", pkg))
+			}
+		}
+	}
+}
+
+func (v *AgentValidator) isMaliciousPackage(packageManager, packageName string, maliciousList []string) bool {
+	for _, malicious := range maliciousList {
+		if strings.EqualFold(packageName, malicious) {
+			return true
+		}
+	}
+	return false
+}
+
+func (v *AgentValidator) ValidateInstallPath(projectRoot, agentName string) error {
+	if agentName == "" {
+		return fmt.Errorf("agent name cannot be empty")
+	}
+
+	if !validAgentNameRegex.MatchString(agentName) {
+		return fmt.Errorf("invalid agent name: %s (only alphanumeric, dots, underscores, and hyphens allowed)", agentName)
+	}
+
+	agentPath := filepath.Join(projectRoot, "agents", agentName)
```

Comment:
we need to use the `project.Project` object here and then 

use `theproject.Bundler.AgentConfig.Dir` to get the dynamic source directory for agents

we also have a `util.SafeProjectFilename` for the agentname filename transformer that might be useful or at least you could add a method in that file to create a regex if you just need validation